### PR TITLE
Remove printf from stack

### PR DESCRIPTION
**Pull request by pyhys**
Rename printing stack functions to xxx_show()

### CODE REVIEW

File: doc/implementation_details.rst [hunk 1]
```diff
@@ -42,6 +42,15 @@ The header file ``pnet_export.h`` is generated by cmake. It ends up in
 ``build/include/pnet_export.h``.
 
 
+Stack developent helper functions
```
Comment (hefloryd) [inline doc/implementation_details.rst:45]: Stack development

File: include/pnet_api.h [hunk 2]
```diff
@@ -1725,6 +1725,16 @@ PNET_EXPORT int pnet_diag_remove(
  *     0x2000              | Show CFG information.
  *     0x4000              | Show scheduler information.
  *     0x8000              | Show I&M data.
+ *
```
Comment (hefloryd) [inline include/pnet_api.h:1728]: Was this intended? I can't tell from the commit message.

File: src/device/pf_cmdev.c [hunk 3]
```diff
@@ -353,7 +353,7 @@ void pf_cmdev_free_diag(
    }
    else
    {
-      printf("ix =%u\n", (unsigned)item_ix);
+      LOG_ERROR(PNET_LOG, "CMDEV(%d): Failed to free diag with index %u\n", __LINE__, (unsigned)item_ix);
```
Comment (hefloryd) [inline src/device/pf_cmdev.c:356]: The error is not propagated here. Would an assert be more appropriate?

File: src/osal/rt-kernel/osal.c [hunk 4]
```diff
@@ -412,12 +412,10 @@ os_buf_t * os_buf_alloc(uint16_t length)
...
    return pbuf_alloc(PBUF_RAW, length, PBUF_POOL);
 }
+
 void os_buf_free(os_buf_t *p)
 {
-   if (pbuf_free(p) != 1)
-   {
-      printf("\n\nPBUF not freed\n");
-   }
+   assert(pbuf_free(p) == 1);
```
Comment (hefloryd) [inline src/osal/rt-kernel/osal.c:418]: Please use ASSERT for rt-kernel (or CC_ASSERT).

File: src/device/pf_cmdev.c (hunk 3 above)
Comment (pyhys) [reply src/device/pf_cmdev.c:356]: You are right, this is a void xxx_free() function. An out-of-range index should be detected elsewhere (if it is a problem). Here is too late. I will remove the print-out.

File: src/osal/rt-kernel/osal.c (hunk 4 above)
Comment (pyhys) [reply src/osal/rt-kernel/osal.c]: Done

File: include/pnet_api.h (hunk 2 above)
Comment (pyhys) [reply include/pnet_api.h:1728]: Updated commit message

File: doc/implementation_details.rst (hunk 1 above)
Comment (pyhys) [reply doc/implementation_details.rst]: Done

**Pull request merged**